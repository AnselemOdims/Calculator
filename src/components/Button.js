import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { selector, sym } = this.props;
    return (
      <button type="button" className={selector.join(' ')}>{sym}</button>
    );
  }
}

Button.propTypes = {
  selector: PropTypes.arrayOf(PropTypes.string).isRequired,
  sym: PropTypes.string.isRequired,
};

export default Button;
