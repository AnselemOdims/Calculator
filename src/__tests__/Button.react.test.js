/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

test('Button mounts when called', () => {
	const sym = '+';
	const selector = ['digits', 'orange'];
	const handleClick = () => handleClick(sym);
	const component = renderer.create(
		<Button sym={sym} handleBtnClick={handleClick} selector={selector} />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
