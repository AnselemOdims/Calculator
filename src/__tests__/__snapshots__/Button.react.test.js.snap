// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button mounts when called 1`] = `
<button
  className="digits orange"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
